File: CS/GrowthWare.Framework/Model/Profiles/MNameValuePairDetail.cs
using GrowthWare.Framework.Model.Profiles.Base;
using GrowthWare.Framework.Model.Profiles.Interfaces;
using System;
using System.Data;

namespace GrowthWare.Framework.Model.Profiles
{
    /// <summary>
    /// Properties for an Name Value Pair Detail.
    /// </summary>
    [Serializable(), CLSCompliant(true)]
    public class MNameValuePairDetail : MProfile
    {
        #region "Member Properties"
        private int m_NameValuePairSeqId = -1;
        private string m_Text = string.Empty;
        private string m_Value = string.Empty;
        private int m_SortOrder = 0;
        private int m_Status = 1;
        #endregion

        #region "Protected Methods"
        /// <summary>
        /// Initializes the specified dr.
        /// </summary>
        /// <param name="dataRow">The dr.</param>
        protected new void Initialize(DataRow dataRow)
        {
            base.NameColumnName = "STATIC_NAME";
            base.IdColumnName = "NVP_SEQ_DET_ID";
            base.Initialize(dataRow);
            m_NameValuePairSeqId = base.GetInt(dataRow, "NVP_SEQ_ID"); ;
            m_Text = base.Name;
            m_Value = base.GetString(dataRow, "NVP_DET_VALUE");
            m_Status = base.GetInt(dataRow, "STATUS_SEQ_ID");
            m_SortOrder = base.GetInt(dataRow, "SORT_ORDER");
        }
        #endregion

        #region "Public Methods"
        /// <summary>
        /// Provides a new account profile with the default vaules
        /// </summary>
        /// <remarks></remarks>
        public MNameValuePairDetail()
        {
        }

        /// <summary>
        /// Will populate values based on the contents of the data row.
        /// </summary>
        /// <param name="dataRow">Datarow containing base values</param>
        /// <remarks>
        /// Class should be inherited to extend to your project specific properties
        /// </remarks>
        public MNameValuePairDetail(DataRow dataRow)
        {
            this.Initialize(dataRow);
        }
        #endregion

        #region "Public Properties"
        /// <summary>
        /// Gets or sets the Name Value Pair SeqId.
        /// </summary>
        /// <value>The Name Value Pair SeqId.</value>
        public int NameValuePairSeqId
        {
            get { return m_NameValuePairSeqId; }
            set { m_NameValuePairSeqId = value; }
        }

        /// <summary>
        /// Gets or sets the status.
        /// </summary>
        /// <value>The status.</value>
        public int Status
        {
            get { return m_Status; }
            set { m_Status = value; }
        }

        /// <summary>
        /// Gets or sets the sort order.
        /// </summary>
        /// <value>The sort order.</value>
        public int SortOrder
        {
            get { return m_SortOrder; }
            set { m_SortOrder = value; }
        }

        /// <summary>
        /// Gets or sets the text.
        /// </summary>
        /// <value>The text.</value>
        public string Text
        {
            get { return m_Text; }
            set { if (!String.IsNullOrEmpty(value)) m_Text = value.Trim(); }
        }

        /// <summary>
        /// Gets or sets the value.
        /// </summary>
        /// <value>The value.</value>
        public string Value
        {
            get { return m_Value; }
            set { if (!String.IsNullOrEmpty(value)) m_Value = value.Trim(); }
        }
        #endregion
    }
}

File: Net_Framework/VB/GrowthWare.Framework/Model/Profiles/MNameValuePairDetail.vb
Imports GrowthWare.Framework.Model.Profiles.Base
Imports GrowthWare.Framework.Model.Profiles.Interfaces

Namespace Model.Profiles
    ''' <summary>
    ''' Base properties an account Profile you can inherit this class and add
    ''' any properties or methods that suit your project needs.
    ''' </summary>
    ''' <remarks>
    ''' Corresponds to table [ZGWSystem].[Name_Value_Pairs] and the table created to store the
    ''' detail information example [ZGWSystem].[Statuses]
    ''' </remarks>
    <Serializable(), CLSCompliant(True)> _
    Public Class MNameValuePairDetail
        Inherits MProfile

#Region "Member Properties"
        Private m_NameValuePairSeqId As Integer = -1
        Private m_Text As String = String.Empty
        Private m_Value As String = String.Empty
        Private m_SortOrder As Integer = 0
        Private m_Status As Integer = 1
#End Region

#Region "Protected Methods"
        Public Shadows Sub Init(ByVal dataRow As DataRow)
            MyBase.IdColumnName = "NVP_SEQ_DET_ID"
            MyBase.NameColumnName = "NVP_DET_TEXT"
            MyBase.Initialize(dataRow)
            m_NameValuePairSeqId = MyBase.GetInt(dataRow, "NVP_SEQ_ID")
            m_Text = MyBase.Name
            m_Value = MyBase.GetString(dataRow, "NVP_DET_VALUE")
            m_Status = MyBase.GetInt(dataRow, "STATUS_SEQ_ID")
            m_SortOrder = MyBase.GetInt(dataRow, "SORT_ORDER")
        End Sub
#End Region

#Region "Public Methods"
        ''' <summary>
        ''' Provides a new account profile with the default vaules
        ''' </summary>
        ''' <remarks></remarks>
        Public Sub New()

        End Sub

        ''' <summary>
        ''' Will populate values based on the contents of the data row.
        ''' </summary>
        ''' <param name="dataRow">Datarow containing base values</param>
        ''' <remarks>
        ''' Class should be inherited to extend to your project specific properties
        ''' </remarks>
        Public Sub New(ByVal dataRow As DataRow)
            Init(dataRow)
        End Sub
#End Region

#Region "Public Properties"
        ''' <summary>
        ''' Gets or sets the Name value pair sequence identifier.
        ''' </summary>
        ''' <value>The Name value pair sequence identifier.</value>
        Public Property NameValuePairSeqId() As Integer
            Get
                Return m_NameValuePairSeqId
            End Get
            Set(ByVal value As Integer)
                m_NameValuePairSeqId = value
            End Set
        End Property

        ''' <summary>
        ''' Gets or sets the status.
        ''' </summary>
        ''' <value>The status.</value>
        Public Property Status() As Integer
            Get
                Return m_Status
            End Get
            Set(ByVal value As Integer)
                m_Status = value
            End Set
        End Property

        ''' <summary>
        ''' Gets or sets the sort order.
        ''' </summary>
        ''' <value>The sort order.</value>
        ''' <remarks>Default value is 0</remarks>
        Public Property SortOrder() As Integer
            Get
                Return m_SortOrder
            End Get
            Set(ByVal value As Integer)
                m_SortOrder = value
            End Set
        End Property

        ''' <summary>
        ''' Gets or sets the text.
        ''' </summary>
        ''' <value>The text.</value>
        Public Property Text() As String
            Get
                Return m_Text
            End Get
            Set(ByVal Value As String)
                If Not Value Is Nothing Then m_Text = Value.Trim
            End Set
        End Property

        ''' <summary>
        ''' Gets or sets the value.
        ''' </summary>
        ''' <value>The value.</value>
        Public Property Value() As String
            Get
                Return m_Value
            End Get
            Set(ByVal Value As String)
                If Not Value Is Nothing Then m_Value = Value.Trim
            End Set
        End Property
#End Region
    End Class
End Namespace
